src/auth/README.md

```diff
@@ -72,69 +83,196 @@ if (blockstack.isUserSignedIn()) {
+    domain_name, // app origin
+    manifest_uri, // url to manifest file - must be hosted on app origin
+    redirect_uri, // url to which browser redirects user on auth approval - must be hosted on app origin
+    version, // version tuple
+    do_not_include_profile, // a boolean flag asking browser to send profile url instead of profile object
+    supports_hub_url, // a boolean flag indicating gaia hub support
+    scopes // an array of string values indicating scopes requested by the app
+  }
 ```
+ 
+  
+### Authentication response payload schema
+  
+```JavaScript
+    const responsePayload = {
+    jti, // UUID
+    iat, // JWT creation time in seconds
+    exp, // JWT expiration time in seconds
+    iss, // legacy decentralized identifier (string prefix + identity address) - this uniquely identifies the user
+    private_key, // encrypted private key payload
+    public_keys, // single entry array with public key
+    profile, // profile object or null if passed by profile_url
+    username, // blockstack id username (if any)
+    core_token, // encrypted core token payload
+    email, // email if email scope is requested & email available
+	  profile_url, // url to signed profile token
+    hubUrl, // url pointing to user's gaia hub
+    version // version tuple
+  }
+```
+  
+## `blockstack:` custom protocol handler
+
+The `blockstack:` custom protocol handler is how Blockstack apps send their authentication requests to the Blockstack Browser. When the Blockstack Browser is installed on a user's computer, it registers itself as the handler for the `blockstack:` customer protocol.
+
+When apps call [`redirectToSignIn`](http://blockstack.github.io/blockstack.js/index.html#redirecttosignin) or [`redirectToSignInWithAuthRequest`](http://blockstack.github.io/blockstack.js/index.html#redirecttosigninwithauthrequest), the code tries to check if a handler for the `blockstack:` custom protocol is installed and, if so, redirects the user to `blockstack:<authRequestToken>`. This causes the authentication request token to be passed from the app to the Blockstack Browser which in turn validates the request, displays information about the app to the user and asks the user if she wants to approve sign in to the app.
 
-### Verify an auth response
+## Adding Blockstack Authentication to your app
+
+The way you can add Blockstack Authentication to you app depends on whether your app is a modern decentralized Blockstack App where code runs client-side without trusted servers or a legacy client-server app where a server is trusted. 
+
+### Authentication in Client-side apps
+This method is appropriate for decentralized client-side apps where the user's zone of trust - the parts of the app that the user is trusting - begins and ends with the code running on their own computer. In apps like these, any code the app interacts with that's not on their own computer such as external servers does not need to know who she is.
```

Comment:
"In apps like these, any code the app interacts with that's not on their own computer such as external servers does not need to know who she is." ->
"In apps like these, any code the app interacts with _not on the user's computer_ should not need to know who the user is."